### ISSUE TITLE
Modin not fully utilize CPU

### ISSUE THREAD

**Issue by thangbn** [modin-project/modin#828]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 18.04
Linux ip-172-30-0-250 4.4.0-1095-aws #106-Ubuntu SMP Wed Sep 18 13:33:48 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- **Modin installed from (source or binary)**:
binary
- **Modin version**:
0.6.1
- **Python version**:
3.7
- **Exact command to reproduce**:
Code is run from Jupyter notebook
### Describe the problem
It always happen that only one CPU getting used fully.
![Screen Shot 2019-10-04 at 5 06 50 PM](https://user-images.githubusercontent.com/5661982/66246495-63482c00-e6c9-11e9-8836-7242bf4db008.png)

What is super weird is exact the same code running a bigger machine (32 CPUS) will be parallelized just fine.
### Source code / logs
```
import modin.pandas as pd
df = pd.read_csv('../data_cleaned/guestbook/storycontributionOct4th.csv',error_bad_lines=False,engine='python', encoding='utf-8')
print(df.shape)
def extract_json (row):
    try:        
        jsondata = json.loads(row['DataJSON'])
        row['firstname'] = jsondata.get('Firstname')
        
        content = jsondata.get('Content')
        row['content'] = content        
    except Exception as e:
        raise e
    return row

df4 = df2.apply(extract_json, axis=1)
```

**Comment by devin-petersohn** [comment]
Hi @thangbn, thanks for the report! I have some follow up questions if you could help me understand a little more:

How long is the utilization this low? Do you know which command was running when the utilization was low? Is it near the beginning or end of the run?

We have some good straggler mitigation techniques internally, so I would definitely like to understand what is happening here.

**Comment by thangbn** [comment]
@devin-petersohn thanks for your reply

the utilization is low since it steps into this line which running the apply **func** on the DataFrame
```
df4 = df2.apply(extract_json, axis=1)
```
and staying low for the next 5 mins, after that I just cancelled the task.
To give you more context: the input file have around 1 million+ rows

**Comment by devin-petersohn** [comment]
Thanks @thangbn, would you be able to provide some sample data? I would like to dig into the utilization problem, but it is difficult without the ability to reproduce the behavior.

**Comment by thangbn** [comment]
@devin sure thing. find the small part extracted from the original file attached.

[extract.txt](https://github.com/modin-project/modin/files/3699093/extract.txt)

**Comment by thangbn** [comment]
@devin-petersohn  just found out downgrade to 0.5.4 work just fine

**Comment by thangbn** [comment]
closed as no longer reproduced on 0.6.2
